### PR TITLE
tests: add and improve some tests

### PR DESCRIPTION
**Pull request by EdwardSro**
This series adds a new SW Steering test and improve ODP tests by explicitly invalidating the MRs to make sure a page fault occurs.
In addition, adjusts the way we get the net-interface to be dependent on both the device name and port.

### CODE REVIEW

File: tests/base.py [hunk 1]
```diff
@@ -180,11 +180,18 @@ def is_eth_and_has_roce_hw_bug(self):
...
-                                       '/sys/class/infiniband/{}/device/net/'
-                                      .format(dev)])
-        return out.decode().split('\n')[0]
+    def get_net_name(dev, port=None):
+        if port is not None:
+            out = subprocess.check_output(['rdma', 'link',  'show', '-j'])
+            loaded_json = json.loads(out.decode())
+            for row in loaded_json:
+                if row['ifname'] == dev and row['port'] == port:
+                    net_name = row['netdev']
```
Comment (mrgolin) [inline tests/base.py:189]: This breaks all pyverbs tests for EFA provider.

Comment (rleon) [reply tests/base.py:189]: Why? Can you please provide a fix to this get_net_name() before release of rdma-core?

Comment (mrgolin) [reply tests/base.py:189]: Sure. Already have a fix, will send it shortly.

Comment (mrgolin) [reply tests/base.py:189]: Please see [this PR](https://github.com/linux-rdma/rdma-core/pull/1328).

**Pull request merged**